homeassistant/components/thermostat/zwave.py

```diff
@@ -40,16 +40,15 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     if (value.node.manufacturer_id.strip() and
             value.node.product_id.strip()):
         specific_sensor_key = (int(value.node.manufacturer_id, 16),
-                               int(value.node.product_id, 16),
-                               value.index)
+                               int(value.node.product_id, 16))
         if specific_sensor_key in DEVICE_MAPPINGS:
             if DEVICE_MAPPINGS[specific_sensor_key] == WORKAROUND_IGNORE:
                 _LOGGER.debug("Remotec ZXT-120 Zwave Thermostat, ignoring")
                 return
-    else:
-        add_devices([ZWaveThermostat(value)])
-        _LOGGER.debug("discovery_info=%s and zwave.NETWORK=%s",
-                      discovery_info, zwave.NETWORK)
+        else:
```

Comment:
This is not going to work. Now it will not add any devices unless there is a manufacturer_id and product_id.

Instead, adding devices should not be part of any `else` at all and be at the same level as `if (value.node…`.